Add support for python 3.11

Numba supports python 3.11 since 0.57.0. Now, that #1030 is merged we can provide this feature to numba-dpex.

I've faced following issue during build:
```
#18 9.187 gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include/python3.11 -I/usr/local/lib/python3.11/site-packages -I/usr/local/lib/python3.11/site-packages/numpy/core/include -I/usr/local/lib/python3.11/site-packages/dpctl/include -I/usr/local/include/python3.11 -c numba_dpex/core/runtime/_dpexrt_python.c -o build/temp.linux-x86_64-cpython-311/numba_dpex/core/runtime/_dpexrt_python.o
#18 9.328 In file included from numba_dpex/core/runtime/_dpexrt_python.c:20:
#18 9.328 numba_dpex/core/runtime/_nrt_python_helper.h:40:2: error: #error "Python minor version is not supported."
#18 9.328    40 | #error "Python minor version is not supported."
#18 9.328       |  ^~~~~
#18 9.344 error: command '/usr/bin/gcc' failed with exit code 1
```

--- a/numba_dpex/core/runtime/_nrt_python_helper.c
+++ b/numba_dpex/core/runtime/_nrt_python_helper.c
@@ -183,7 +183,7 @@ PyTypeObject MemInfoType = {
  * act as a reminder to update this struct on Python version update! */
 #if (PY_MAJOR_VERSION == 3)
 #if !((PY_MINOR_VERSION == 8) || (PY_MINOR_VERSION == 9) ||                    \
-      (PY_MINOR_VERSION == 10))
+      (PY_MINOR_VERSION == 10) || (PY_MINOR_VERSION == 11))
 #error "Python minor version is not supported."
 #endif
 #else

--- a/numba_dpex/core/runtime/_nrt_python_helper.h
+++ b/numba_dpex/core/runtime/_nrt_python_helper.h
@@ -32,16 +32,4 @@ PyObject *MemInfo_get_refcount(MemInfoObject *self, void *closure);
 PyObject *MemInfo_get_external_allocator(MemInfoObject *self, void *closure);
 PyObject *MemInfo_get_parent(MemInfoObject *self, void *closure);
 
-/* WARNING: Do not remove this, only modify it! It is a version guard to
- * act as a reminder to update this struct on Python version update! */
-#if (PY_MAJOR_VERSION == 3)
-#if !((PY_MINOR_VERSION == 8) || (PY_MINOR_VERSION == 9) ||                    \
-      (PY_MINOR_VERSION == 10))
-#error "Python minor version is not supported."
-#endif
-#else
-#error "Python major version is not supported."
-#endif
-/* END WARNING*/
-
 #endif /* _NRT_PYTHON_HELPER_H_ */

